File: frontend/src/types/accessCard.ts
export interface AccessCardSummaryDto {
  id: string
  siteId: string
  userId: string
  cardNumber: string
  isActive: boolean
  issueDate: string
  expiryDate?: string
  notes?: string
  createdAt: string
}

export interface CreateAccessCardDto {
  userId: string
  cardNumber: string
  issueDate: string
  expiryDate?: string
  notes?: string
}

export interface UpdateAccessCardDto {
  cardNumber: string
  isActive: boolean
  issueDate: string
  expiryDate?: string
  notes?: string
}

File: backend/src/SiteYonetimi.Infrastructure/Entities/Shared/AccessCard.cs
namespace SiteYonetimi.Infrastructure.Entities.Shared;

public class AccessCard
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid SiteId { get; set; }
    public Guid UserId { get; set; }
    public string CardNumber { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
    public DateTime IssueDate { get; set; } = DateTime.UtcNow;
    public DateTime? ExpiryDate { get; set; }
    public string? Notes { get; set; }
    public bool IsDeleted { get; set; } = false;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }
}
